filebeat/input/file.go

```diff
@@ -75,6 +85,29 @@ func (f *FileEvent) ToMapStr() common.MapStr {
 		"count":      1,
 	}
 
+	if f.jsonDecoder != nil {
+		var jsonObj common.MapStr
+		err := json.Unmarshal([]byte(*f.Text), &jsonObj)
+		if err != nil {
+			logp.Err("Error decoding JSON: %v", err)
+			if f.jsonDecoder.OnUnmarshalError == "add_error_key" {
+				event["json_error"] = fmt.Sprintf("Error decoding JSON: %v", err)
```

Comment:
I'm not sure if there are cases with multiple errors as I assume every time we hit an error we would stop executing the next steps and send the event with the error. For example if there is a multiline error we should not continue with JSON decoding.